File: DSA/YashKrishanGupta_590014798/Day_29_Question_1.java
import java.util.Scanner;
public class Day_29_Question_1{
    // Solution 1
    public static void checkEvenOrOdd(int n){
        if((n & 1) == 0) System.out.print("Number is even");
        else System.out.print("Number is odd");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number : ");
        int num = sc.nextInt();
        checkEvenOrOdd(num);
    }
}

File: DSA/Rahul_Mankani_590014725/Day29_Question1.c
#include <stdio.h>

int even(int n) 
{
    // If n & 1 is 0 - number is even, if 1 - number is odd
    return (n & 1) == 0;
}
int main() 
{
    int n; 
    printf("Enter a number: ");
    scanf("%d", &n);
    if (even(n)) 
    {
        printf("%d is even\n", n);
    } 
    else 
    {
        printf("%d is odd\n", n);
    }
    return 0;
}
